completion/tests/test_services.py

```diff
@@ -80,6 +84,67 @@ def test_enabled_honors_waffle_switch(self, enabled):
         with self.override_completion_switch(enabled):
             self.assertEqual(self.completion_service.completion_tracking_enabled(), enabled)
 
+    @ddt.data(
+        (XBlockCompletionMode.COMPLETABLE, False, False, True),
+        (XBlockCompletionMode.COMPLETABLE, True, False, False),
+        (XBlockCompletionMode.COMPLETABLE, False, True, False),
+        (XBlockCompletionMode.AGGREGATOR, False, False, False),
+        (XBlockCompletionMode.EXCLUDED, False, False, False)
+    )
+    @ddt.unpack
+    def test_can_mark_block_complete_on_view(self, mode, has_score, has_custom_completion, can_mark_complete):
+        block = XBlock(Mock(), scope_ids=Mock(spec=ScopeIds))
+        block.completion_mode = mode
+        block.has_score = has_score
+        if has_custom_completion:
+            block.has_custom_completion = True
+
+        self.assertEqual(self.completion_service.can_mark_block_complete_on_view(block), can_mark_complete)
+
+    def test_blocks_to_mark_complete_on_view(self):
+
+        block_1 = XBlock(Mock(), scope_ids=Mock(spec=ScopeIds))
+        block_1.location = UsageKey.from_string("i4x://edX/100/a/1").replace(course_key=self.course_key)
```

Comment:
Is this use of replace the same as `usage_key.map_into_course(course_key)`?

Reply:
I suspect so. I just copied this pattern from line 37 above.